cache_path.go

```diff
@@ -201,26 +204,27 @@ func match(pth string, excludeByPattern map[string]bool) (bool, bool) {
 // Otherwise a path will affect the previous cache invalidation:
 // if the path has indicator, the indicator will affect the previous cache invalidation
 // otherwise the file itself.
-func interleave(indicatorByPth map[string]string, excludeByPattern map[string]bool) (map[string]string, error) {
-	indicatorByCachePth := map[string]string{}
+func interleave(indicatorMap map[string][]string, excludeByPattern map[string]bool) (map[string][]string, error) {
+	indicatorMapByCache := map[string][]string{}
+
+	for indicator, pths := range indicatorMap {
+		for _, pth := range pths {
+			skip, exclude := match(pth, excludeByPattern)
+			if exclude {
+				// this file should not be included in the cache
+				continue
+			}
 
-	for pth, indicator := range indicatorByPth {
-		skip, exclude := match(pth, excludeByPattern)
-		if exclude {
-			// this file should not be included in the cache
-			continue
+			if skip || indicator == "-" {
+				// this file's changes does not fluctuates existing cache invalidation
+				indicator = ""
+			} else if len(indicator) == 0 {
+				// the file's own content fluctuates existing cache invalidation
+				indicator = pth
```

Comment:
Mutating the map key has unwanted effects. Consider the test case:
```	{
			name:                "no indicator, own content is the indicator",
			indicatorMap:        map[string][]string{"": {"path/to/cache", "path/to/cache2"}},
			excludeByPattern:    map[string]bool{},
			indicatorByCachePth: map[string][]string{"path/to/cache": {"path/to/cache"}, "path/to/cache2": {"path/to/cache2"}},
			wantErr:             false,
		},
        ```

Reply:
Updated code.